### PR TITLE
PMM-11808 Fix query results deserialization bug

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-11808

Link to the Feature Build: SUBMODULES-3142

This PR reverts some changes related to data transferring introduced in https://github.com/percona/pmm/pull/1814
Also it adds base64 encoding for binary date before json serialization.

### CODE REVIEW

File: managed/services/checks/checks.go [hunk 1]
```diff
@@ -1266,8 +1279,18 @@ WHERE datallowconn = true AND datistemplate = false AND has_database_privilege(c
...
 
+	dec, err := b64.DecodeString(res)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode database discovery results")
+	}
+
+	data, err := agentpb.UnmarshalActionQueryResult(dec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshall database discovery results")
+	}
```
Comment (artemgavrilov) [inline managed/services/checks/checks.go:1290]: Inconvenient part, but I will think how we can eliminate it later

File: managed/cmd/pmm-managed-starlark/main.go [hunk 2]
```diff
@@ -117,17 +119,52 @@ func runChecks(l *logrus.Entry, data *checks.StarlarkScriptData) ([]check.Result
...
+		results, err = env.Run(data.Name, res, contextFuncs, l.Debugln)
 	}
 	if err != nil {
 		return nil, errors.Wrap(err, "error running starlark env")
 	}
 
 	return results, nil
 }
+
+func unmarshallQueryResult(qr string) ([]map[string]any, error) {
```
Comment (idoqo) [inline managed/cmd/pmm-managed-starlark/main.go:158]:
```suggestion
func unmarshalQueryResult(qr string) ([]map[string]any, error) {
```

File: managed/services/checks/checks.go (hunk 1 above)
Comment (idoqo) [inline managed/services/checks/checks.go:1290]: This block is similar to the `unmarshalQueryResult` function in `main.go`. I wonder if we can export bring the function to this package and export it for `main.go` to use.

**Review by idoqo — APPROVED** [review]

File: managed/services/checks/checks.go (hunk 1 above)
Comment (artemgavrilov) [reply managed/services/checks/checks.go:1290]: I will address these comments in another PR. For now we need just to fix RC ASAP.

**Review by PavelKhripkov — APPROVED** [review]

File: managed/cmd/pmm-managed-starlark/main.go [hunk 3]
```diff
@@ -117,17 +119,55 @@ func runChecks(l *logrus.Entry, data *checks.StarlarkScriptData) ([]check.Result
...
+	b, err := base64.StdEncoding.DecodeString(qr)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode base64 encoded query result")
+	}
+
+	res, err := agentpb.UnmarshalActionQueryResult(b)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshall query result")
```
Comment (atymchuk) [inline managed/cmd/pmm-managed-starlark/main.go:169]:
```suggestion
		return nil, errors.Wrap(err, "failed to unmarshal query result")
```
There should be a single `l` everywhere )

File: managed/services/checks/checks.go [hunk 4]
```diff
@@ -1266,8 +1279,18 @@ WHERE datallowconn = true AND datistemplate = false AND has_database_privilege(c
...
-	r := make([]string, len(res))
-	for i, row := range res {
+	dec, err := b64.DecodeString(res)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode database discovery results")
+	}
+
+	data, err := agentpb.UnmarshalActionQueryResult(dec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshall database discovery results")
```
Comment (atymchuk) [inline managed/services/checks/checks.go:1289]:
```suggestion
		return nil, errors.Wrap(err, "failed to unmarshal database discovery results")
```

**Review by atymchuk — APPROVED** [review]

**Pull request merged**